### PR TITLE
Statefulsets

### PR DESCRIPTION
**Pull request by markyjackson-taulia**
1. Corrected statefulsets yaml
2. Added note about statefulsets and k8s version before 1.9 not being stable
3. Added sysdigcloud yamls directory

Please review @mbreitung

### CODE REVIEW

**Comment by mbreitung** [comment]
the yamls should be 0644 and not 0755

File: backend/sdc-api.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+apiVersion: v1
+kind: Service
```
Comment (mbreitung) [inline backend/sdc-api.yaml]: Can we talk about splitting the yaml files where each file has a single purpose? i.e.  sdc-api-service.yaml and sdc-api-deployment.yaml vs having both service and deployment in the same file

File: backend/sdc-collector.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            - name: API_ENDPOINT
+              valueFrom:
+                configMapKeyRef:
+                  name: sysdigcloud-config
+                  key: api.url
+            - name: JVM_EXTRA_OPTS
+              valueFrom:
+                configMapKeyRef:
+                  name: sysdigcloud-config
+                  key: sysdigcloud.jvm.options
```
Comment (mbreitung) [inline backend/sdc-collector.yaml]: all jvms share the same jvm config?

File: backend/sdc-api.yaml (hunk 1 above)
Comment (markyjackson-taulia) [reply backend/sdc-api.yaml]: Sure, lets chat. It is common to bundle up the deployment with associated sac

File: backend/sdc-worker.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            - name: logs
+              mountPath: /var/log/sysdigcloud
+            - name: config
+              mountPath: /opt/kubernets_config
+            - name: ssl
+              mountPath: /opt/certs/
+          resources:
+            requests:
+              cpu: 1000m
+              memory: 3Gi
```
Comment (mbreitung) [inline backend/sdc-worker.yaml]: Can we talk about either different request/limit or making this a template

File: bin/create-manifests.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+SDC_HOME="../"
+SDC_CONFIG="$SDC_HOME/etc/config"
+SDC_TEMPLATES="$SDC_CONFIG/templates"
+SDC_SETTINGS_FILE="$SDC_CONFIG/sdc-settings.yaml"
+PVC_TYPE="$(egrep 'storageclassName'  $SDC_SETTINGS_FILE |egrep -v '#'|awk -F: '{print $2}')"
```
Comment (mbreitung) [inline bin/create-manifests.sh]: egrep is not needed here

File: backend/sdc-collector.yaml (hunk 2 above)
Comment (markyjackson-taulia) [reply backend/sdc-collector.yaml]: In this iteration, yes

File: bin/create-manifests.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+SDC_HOME="../"
```
Comment (mbreitung) [inline bin/create-manifests.sh]: ouch `../`?

File: bin/create-manifests.sh [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+PVC_TYPE="$(egrep 'storageclassName'  $SDC_SETTINGS_FILE |egrep -v '#'|awk -F: '{print $2}')"
+
+
+#create manifests for configmaps
+kontemplate template $SDC_SETTINGS_FILE -i templates/configmaps/ > $SDC_HOME/etc/config/sdc-config.yaml
+
+#create manifests for datastores
+
+if [ $PVC_TYPE == "gce-pd" ]; then
+	kontemplate template $SDC_SETTINGS_FILE -i templates/datastores/storageclasses/gce-pd/ > $SDC_HOME/datastores/storageclasses/sdc-storageclass.yaml
```
Comment (mbreitung) [inline bin/create-manifests.sh]: could this be 4 spaces instead of 8?

File: bin/install.sh [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+NOW=$(date "+%Y.%m.%d-%H.%M.%S")
+SDC_HOME=".."
+LOG_FILE=$SDC_HOME/logs/install/install-$NOW.log
+SETTINGS_FILE="$SDC_HOME/etc/config/sdc-settings.yaml"
+NAMESPACE=$(egrep namespace $SETTINGS_FILE |awk '{print $2}')
+BACKEND_VERSION=$(egrep backendImage $SETTINGS_FILE | awk -F: '{print $3}')
+FRONTEND_VERSION=$(egrep frontendImage $SETTINGS_FILE | awk -F: '{print $3}')
+K8S_CLIENT_VERSION=$(kubectl version | egrep ^Client | awk -F, '{print $3}')
+K8S_SERVER_VERSION=$(kubectl version | egrep ^Server | awk -F, '{print $3}')
+CURRENT_CONTEXT=$(kubectl config get-contexts|egrep '^\*'|awk '{print $2}')
```
Comment (mbreitung) [inline bin/install.sh]: please be consistent with spaces around `|` for instance

File: bin/create-manifests.sh [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+SDC_HOME="../"
+SDC_CONFIG="$SDC_HOME/etc/config"
+SDC_TEMPLATES="$SDC_CONFIG/templates"
+SDC_SETTINGS_FILE="$SDC_CONFIG/sdc-settings.yaml"
+PVC_TYPE="$(egrep 'storageclassName'  $SDC_SETTINGS_FILE |egrep -v '#'|awk -F: '{print $2}')"
+
+
+#create manifests for configmaps
+kontemplate template $SDC_SETTINGS_FILE -i templates/configmaps/ > $SDC_HOME/etc/config/sdc-config.yaml
```
[the remark this answers is not in the document]
Comment (markyjackson-taulia) [reply bin/create-manifests.sh]: Assumption is it is already installed. In readme: - [kontemplate] (https://github.com/tazjin/kontemplate) is required for templatized deployment.

File: backend/sdc-api.yaml (hunk 1 above)
Comment (mbreitung) [reply backend/sdc-api.yaml]: agreed, this is ok for now

File: backend/sdc-collector.yaml (hunk 2 above)
Comment (mbreitung) [reply backend/sdc-collector.yaml]: fair, but we should change that in the future

File: backend/sdc-worker.yaml (hunk 3 above)
Comment (mbreitung) [reply backend/sdc-worker.yaml]: we should discuss sane values for this. What are the default jvm params?

File: datastores/sdc-redis-slaves.yaml [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        role: redis-slave
+  template:
+    metadata:
+      labels:
+        app: sysdigcloud
+        role: redis-slave
+    spec:
+      containers:
+      - name: redis-cluster
+        image: yofti/redis:4.0.2
```
Comment (mbreitung) [inline datastores/sdc-redis-slaves.yaml]: yofti/redis?

Comment (markyjackson-taulia) [reply datastores/sdc-redis-slaves.yaml]: Hahaha, missed that. I’ll remove that and get the approved ver

**Comment by markyjackson-taulia** [comment]
@mbreitung All yours

File: README.md [hunk 10]
```diff
@@ -1,195 +1,384 @@
-# Sysdig Cloud on Kubernetes
+# sdc-kubernetes: Sysdig Cloud Monitor Backend on Kubernetes
```
Comment (mbreitung) [inline README.md]: why this wording change?

File: README.md [hunk 11]
```diff
@@ -1,195 +1,384 @@
...
 
-- Running Kubernetes cluster, Kubernetes version >= 1.3.X (this guide has been tested with Kubernetes 1.3.6)
-- Sysdig Cloud quay.io pull secret
-- Sysdig Cloud license
 
-### Infrastructure Overview
+## What is this? <a id="What-is-this?"></a>
 
-![Sysdig Cloud infrastructure](images/sysdig_cloud_infrastructure.png?raw=true "Infrastructure")
+sdc-kubernetes is an on-prem version of [Sysdig Monitor](https://sysdig.com/product/monitor/), a SAAS offering by Sysdig Inc for monitoring containerized envrionments. The official on-prem Kubernetes guide can be found [here](https://github.com/draios/sysdigcloud-kubernetes). This repo is the result of a personal, on-going proof-of-concept project on improving certain aspects of Kubernetes deployment.
```
Comment (mbreitung) [inline README.md]: Wording needs updated here

Comment (markyjackson-taulia) [reply README.md]: 👍

File: etc/config/templates/frontend/sdc-agent-daemonset-template.yaml [hunk 12]
```diff
@@ -0,0 +1,97 @@
+#Use this sysdig.yaml when Daemon Sets are enabled on Kubernetes (minimum version 1.1.1). Otherwise use the RC method.
```
Comment (smartinovic) [inline etc/config/templates/frontend/sdc-agent-daemonset-template.yaml]: Not sure if this is scope of this PR but we should update agent daemonset to work with RBAC (as in newer versions of k8s RBAC is enabled by default) as described here https://support.sysdig.com/hc/en-us/articles/206770633

Comment (markyjackson-taulia) [reply etc/config/templates/frontend/sdc-agent-daemonset-template.yaml]: We can look at that in the next pass and I will add that to the TO DO list. This PR is just for the introduction of statefulsets.

Comment (mbreitung) [reply etc/config/templates/frontend/sdc-agent-daemonset-template.yaml]: even more, we should not have this code here, but pull it from the agent repo so we don't need to keep this file up-to-date

**Pull request merged**